### PR TITLE
refactor(module:calendar): provide new implementation

### PR DESCRIPTION
**Pull request by trotyl**
Depends on radio(done), select(wip).

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

### CODE REVIEW

File: components/calendar/doc/index.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+按照日历形式展示数据的容器。
+
+## 何时使用
+
+当数据是日期或按照日期划分时，例如日程、课表、价格日历等，农历等。目前支持年/月切换。
+
+## API
+
+**注意：**Calendar 部分 locale 是从 value 中读取，所以请先正确设置 moment 的 locale。
```
Comment (vthinkxie) [inline components/calendar/doc/index.zh-CN.md]: moment is not needed?

File: components/calendar/nz-calendar-header.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<nz-select class="ant-fullcalendar--year-select" [nzSize]="size"
+           [(ngModel)]="activeYear" (ngModelChange)="yearChange.emit($event)">
```
Comment (vthinkxie) [inline components/calendar/nz-calendar-header.component.html]: should [ngModel], there is no activeYear set function.

File: components/calendar/nz-calendar-header.component.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  <nz-option *ngFor="let year of years" [nzLabel]="year.label" [nzValue]="year.value"></nz-option>
+</nz-select>
+
+<nz-select *ngIf="mode === 'month'" class="ant-fullcalendar--year-select" [nzSize]="size" 
+           [(ngModel)]="activeMonth" (ngModelChange)="monthChange.emit($event)">
```
Comment (vthinkxie) [inline components/calendar/nz-calendar-header.component.html]: same problem

File: components/calendar/nz-calendar-header.component.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  @Output() yearChange: EventEmitter<number> = new EventEmitter();
+  @Output() monthChange: EventEmitter<number> = new EventEmitter();
+
+  yearOffset: number = 10;
+  yearTotal: number = 20;
+  years: Array<{label: string, value: number}>;
+  months: Array<{label: string, value: number}>;
+
+  get activeYear(): number {
```
Comment (vthinkxie) [inline components/calendar/nz-calendar-header.component.ts]: should add set activeYear to update optionList year range.

File: components/calendar/nz-calendar-header.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<nz-select class="ant-fullcalendar--year-select" [nzSize]="size"
```
Comment (vthinkxie) [inline components/calendar/nz-calendar-header.component.html]: the className should be changed when the nzFullscreen change

Comment (trotyl) [reply components/calendar/nz-calendar-header.component.html]: This class is static in current react implementation

Comment (vthinkxie) [reply components/calendar/nz-calendar-header.component.html]: className should be `ant-fullcalendar-year-select`

File: components/calendar/nz-calendar-header.component.html (hunk 3 above)
Comment (vthinkxie) [reply components/calendar/nz-calendar-header.component.html]: className should be `ant-fullcalendar-month-select`

**Pull request merged**